TeaLeaf/src/com/tealeaf/TeaLeaf.java

```diff
@@ -362,6 +370,32 @@ public void onGlobalLayout(){
 		});
 	}
 
+	private boolean isNotchedDevice() {
+		String[] notched_models = { "ANE-AL00", "ANE-L21", "ANE-TL00", "ANE-LX1", "ANE-LX2", "ANE-LX2J", "ANE-LX3", //Huawei nova 3e
+				"CLT-L04", "CLT-L09", "CLT-L29", // Huawei P20 Pro
+				"EML-L09", "EML-L22", "EML-L29", // Huawei P20
+				"ANE-L02", "ANE-L02K", "ANE-L03", "ANE-L12JPZ", "ANE-L21", "ANE-L22", "ANE-L23", // Huawei P20 Lite
+				"ZE620KL", // Asus ZenFone 5
+				"COL-AL00", "COL-AL10", "COL-TL00", "COL-TL10", // Huawei Honor 10
+				"R15 Pro", "PAAM00", // Oppo R15 Pro
+				"R15", // Oppo R15
+				"CPH1819", // Oppo F7
+				"V9", // Vivo v9
+				"V9 Youth", // Vivo v9 Youth
+				"X21", "X21 UD",  // Vivo X21 and X21 UD
+				"A6000", "A6003", // One Plus 6
+				"LM-G710", "G710" }; // LG G7 ThinQ
+
+		boolean notched_device = false;
+
+		for (String notched_model : notched_models) {
+			if (Build.MODEL.equals(notched_model)) {
+				notched_device = true;
+			}
```

Comment:
this can be written better for sure. We should return either immediately or use a better datastructure, so we can use built in functions to check if it is present